adapters/http-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java

```diff
@@ -631,28 +627,27 @@ private void doUploadMessage(
                     return getTimeUntilDisconnect(tenantTracker.result(), ttdParam)
                             .onSuccess(effectiveTtd -> Optional.ofNullable(effectiveTtd)
                                     .ifPresent(v -> TracingHelper.TAG_DEVICE_TTD.set(currentSpan, v)));
-                });
-        final Future<CommandConsumer> commandConsumerTracker = ttdTracker
-                .compose(ttd -> createCommandConsumer(
-                        ttd,
-                        tenantTracker.result(),
-                        deviceId,
-                        gatewayId,
-                        ctx.getRoutingContext(),
-                        responseReady,
-                        currentSpan));
-
-        commandConsumerTracker
-        .compose(ok -> {
+                })
+                .map(ttd -> ttd == null || ttd <= 0 ? null : ttd); // non positive ttd are made null - no ttd
+
+        ttdTracker
+        .compose(ttd -> createResponseReadyTracker(
+                ttd,
+                tenantTracker.result(),
+                deviceId,
+                gatewayId,
+                ctx.getRoutingContext(),
+                currentSpan))
+        .compose(responseReadyTracker -> {
```

Comment:
Looking at the code here and seeing the method name `createResponseReadyTracker`, it doesn't become obvious that this is about handling of Command & Control messages.
I would therefore suggest keeping the `createCommandConsumer` method name.

Reply:
Actually my idea was that this entity tracks response readiness to be send. And actually reflects the returned type.
I think maybe `waitForCommand` should be renamed to just `waitFor`. WDYT?

Reply:
> Actually my idea was that this entity tracks response readiness to be send. And actually reflects the returned type.

FMPOV, the name `createResponseReadyTracker` is rather abstract while `createCommandConsumer` gives a better indication what the method is doing (especially that it's about *command* messages). The idea that the method return value tracks response readiness is already shown by the return type. (And as the method is not a getter method, I don't think it needs to be named according to its return type.

Reply:
FMPOV this method creates a tracker for response readiness. It hides all complexity of what does it mean, e.g. waiting for command if ttd or immediately ready if not. That's why I don't want to provide any indication for callers of the method.
If you insist I'd change the name to `createCommandConsumer` but I prefer current name.

Reply:
As response readiness is only dependent on command handling here (provided ttd is set), I think it improves readability in `doUploadMessage` if the method name makes this clear.

> waiting for command if ttd or immediately ready if not

The `ttdSecs == null` case and `ResponseReadyTracker.nop()` creation could also be moved out of the method into  `doUploadMessage`. Then `createResponseReadyTracker` only creates a command-related `ResponseReadyTracker`.

Reply:
ok

Reply:
renamed and `ttd == null` moved in `doUploadMessage`